### DISCUSSION TITLE
Failed to run celery as non root user in detached mode in alpine based docker container

### DISCUSSION THREAD

**Discussion by sanket-parte** [celery/celery#8648] [General]
Hi All,

I have followed the following steps to run celery as non-root user in Alpine-based Docker container.

1. Created user and group using the  following command:
```shell
addgroup celery && adduser --disabled-password --gecos "" -S -s /bin/sh celery -G celery -D
``` 
2. Created pid file at "/var/run/project_name/worker-celery.pid"
3. Ran the following command to run celery as above celery user using PID & GID:
```shell
celery --workdir=/app/project_name -A project_name worker -l INFO --logfile=/app/logs/celery.log --hostname=project_name@%h --pidfile=/var/run/project_name/worker-celery.pid --uid=1003 --gid=1003 --detach
``` 
4. After running the celery command. I start a gunicorn command in attached mode to keep the container running. Here I am running the celery along with django project.
5. Despite following these steps, celery is not running in the background and is not logging anything to the log file.

When I ran the celery command in attached mode, it worked fine.

Can anyone help me with this?

**Comment by lyf2000** [comment]
i solved it by adding `&` at the end instead of `--detach`:
```bash
for i in $(seq 1 $DIALOG_QUEUES); do
echo tg.message.v1.$i; 
celery -A core worker -l INFO -Q tg.message.v1.$i -c 1 -f /app/logs/celery.tg.message.logs &
done
```
